lms/djangoapps/courseware/views/views.py

```diff
@@ -1648,3 +1611,20 @@ def get_financial_aid_courses(user):
             )
 
     return financial_aid_courses
+
+
+def redirect_if_needed_to_access_course(request, course):
```

Comment:
I just moved the method as is, but I agree that the name is confusing. I'll come up with something clearer.